packages/apm/src/integrations/tracing.ts

```diff
@@ -271,6 +238,119 @@ export class Tracing implements Integration {
     });
   }
 
+  /**
+   * Pings the heartbeat
+   */
+  private static _pingHeartbeat(): void {
+    Tracing._heartbeat = (setTimeout(() => {
+      Tracing._beat();
+    }, 5000) as any) as number;
+  }
+
+  /**
+   * Checks when entries of Tracing._activities are not changing for 3 beats. If this occurs we finish the transaction
+   *
+   */
+  private static _beat(): void {
+    clearTimeout(Tracing._heartbeat);
+    const keys = Object.keys(Tracing._activities);
+    if (keys.length) {
+      const heartbeatString = keys.reduce((prev: string, current: string) => prev + current);
```

Comment:
Won't it be too fragile?

Reply:
Not sure was looking for something reasonable of "hashing" this whole thing

Reply:
Oooh, it's based on keys, so `id`, not type of activity. It shooooould be good enough for now then.

Reply:
This can be unnecessarily large (it is a string of unbounded length). Plus it sort of relies on key order. Looking for something better...

Reply:
Also, isn't this generating a lot of garbage for the GC every 5s?

Reply:
Might not be it yet, but found this

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy
from https://stackoverflow.com/questions/11578377/detecting-change-in-a-javascript-object